--- srsLTE/srsue/ue.conf.example	2021-03-25 21:21:00.004589632 +1100
+++ srsLTE_modified/srsue/ue.conf.example	2021-03-22 16:20:01.224531000 +1100
@@ -35,7 +35,8 @@ dl_earfcn = 3400
 freq_offset = 0
 tx_gain = 80
 #rx_gain = 40
-
+dl_freq = 5675000000
+ul_freq = 5685000000
 #nof_radios = 1
 #nof_rx_ant = 1
 
@@ -43,7 +44,8 @@ tx_gain = 80
 #     USRP B210: pass num_recv_frames=64,num_send_frames=64
 
 #device_args = auto
-#time_adv_nsamples = auto
+#time_adv_nsamples = 171
+time_adv_nsamples = auto
 #burst_preamble_us = auto
 #continuous_tx     = auto
 
@@ -113,8 +115,8 @@ file_max_size = -1
 #####################################################################
 [usim]
 mode = soft
-algo = xor
-#opc  = 63BFA50EE6523365FF14C1F45F88737D
+algo = milenage
+opc  = 63BFA50EE6523365FF14C1F45F88737D
 k    = 00112233445566778899aabbccddeeff
 imsi = 001010123456789
 imei = 353490069873319
@@ -275,7 +277,7 @@ enable = false
 #snr_ema_coeff       = 0.1
 #snr_estim_alg       = refs
 #pdsch_max_its       = 8    # These are half iterations
-#nof_phy_threads     = 3
+nof_phy_threads     = 4
 #equalizer_mode      = mmse
 #sfo_ema             = 0.1
 #sfo_correct_period  = 10
@@ -286,7 +288,7 @@ enable = false
 #snr_to_cqi_offset   = 0.0
 #interpolate_subframe_enabled = false
 #sic_pss_enabled     = true
-#pregenerate_signals = false
+pregenerate_signals = true
 #pdsch_csi_enabled  = true
 #pdsch_8bit_decoder = false
 #force_ul_amplitude = 0
@@ -302,6 +304,6 @@ enable = false
 #
 #####################################################################
 [general]
-#metrics_csv_enable  = false
-#metrics_period_secs = 1
-#metrics_csv_filename = /tmp/ue_metrics.csv
+metrics_csv_enable  = true
+metrics_period_secs = 10
+metrics_csv_filename = /tmp/ue_metrics.csv
