--- srsLTE/srsenb/src/stack/upper/CMakeLists.txt	2021-03-25 21:20:59.992589519 +1100
+++ srsLTE_modified/srsenb/src/stack/upper/CMakeLists.txt	2021-03-22 16:20:01.244532000 +1100
@@ -19,6 +19,6 @@
 #
 
 file(GLOB SOURCES "*.cc")
-add_library(srsenb_upper STATIC ${SOURCES})
+add_library(srsenb_upper STATIC ${SOURCES} ${SRSENB_EXT_SOURCE})
 install(TARGETS srsenb_upper DESTINATION ${LIBRARY_DIR})
 
