--- srsLTE/cmake/modules/FindMKL.cmake	2021-03-25 21:20:50.704501432 +1100
+++ srsLTE_modified/cmake/modules/FindMKL.cmake	2021-03-22 16:20:01.292532000 +1100
@@ -4,7 +4,9 @@
 #  MKL_INCLUDE_DIRS - The mkl include directories
 #  MKL_LIBRARIES - The libraries needed to use mkl
 #  MKL_DEFINITIONS - Compiler switches required for using mkl
-
+if(BUILD_DDRL)
+MESSAGE(STATUS "DISABLE FindMKL for libtorch" )
+else(BUILD_DDRL)
 find_path(MKL_INCLUDE_DIR
             NAMES mkl.h
             HINTS $ENV{MKL_DIR}/include
@@ -49,5 +51,5 @@ if(MKL_FOUND)
   MESSAGE(STATUS "Found MKL_LIBRARIES: ${MKL_LIBRARIES}" )
   MESSAGE(STATUS "Found MKL_STATIC_LIBRARIES: ${MKL_STATIC_LIBRARIES}" )
 endif(MKL_FOUND)
-
 mark_as_advanced(MKL_INCLUDE_DIR MKL_FFTW_INCLUDE_DIR MKL_LIBRARIES MKL_CORE MKL_ILP MKL_SEQ)
+endif(BUILD_DDRL)
\ No newline at end of file
