--- srsLTE/cmake/modules/SRSLTEbuildinfo.cmake.in	2021-03-25 21:20:59.904588684 +1100
+++ srsLTE_modified/cmake/modules/SRSLTEbuildinfo.cmake.in	2021-03-22 16:20:01.296532000 +1100
@@ -16,6 +16,6 @@ OUTPUT_STRIP_TRAILING_WHITESPACE
 
 message(STATUS "Generating build_info.h")
 configure_file(
-  @CMAKE_SOURCE_DIR@/lib/include/srslte/build_info.h.in
-  @CMAKE_BINARY_DIR@/lib/include/srslte/build_info.h
+  @PROJECT_SOURCE_DIR@/lib/include/srslte/build_info.h.in
+  @PROJECT_BINARY_DIR@/lib/include/srslte/build_info.h
 )
