--- srsLTE/cmake/modules/SRSLTE_install_configs.sh.in	2021-03-25 21:20:59.904588684 +1100
+++ srsLTE_modified/cmake/modules/SRSLTE_install_configs.sh.in	2021-03-22 16:20:01.296532000 +1100
@@ -33,8 +33,8 @@ install_file(){
 
   # Check if config file already exists in location
   if [ -f  $dest_path ]; then
-    echo " - $dest_path already exists. Skipping it."
-    return
+    echo " - $dest_path already exists. Deleting it."
+    rm $dest_path
   fi
   
   # Check if config file exists in source location
