--- srsLTE/srsenb/hdr/stack/rrc/rrc.h	2021-03-25 21:20:59.988589480 +1100
+++ srsLTE_modified/srsenb/hdr/stack/rrc/rrc.h	2021-03-22 16:20:01.256532000 +1100
@@ -156,7 +156,6 @@ public:
   void write_dl_info(uint16_t rnti, srslte::unique_byte_buffer_t sdu);
   void release_complete(uint16_t rnti);
   bool setup_ue_ctxt(uint16_t rnti, LIBLTE_S1AP_MESSAGE_INITIALCONTEXTSETUPREQUEST_STRUCT *msg);
-  bool modify_ue_ctxt(uint16_t rnti, LIBLTE_S1AP_MESSAGE_UECONTEXTMODIFICATIONREQUEST_STRUCT *msg);
   bool setup_ue_erabs(uint16_t rnti, LIBLTE_S1AP_MESSAGE_E_RABSETUPREQUEST_STRUCT *msg);
   bool release_erabs(uint32_t rnti);
   void add_paging_id(uint32_t ueid, LIBLTE_S1AP_UEPAGINGID_STRUCT UEPagingID);
@@ -255,8 +254,6 @@ public:
 
     bool connect_notified;
 
-    bool is_csfb;
-
   private:
     srslte::byte_buffer_pool  *pool;
 
@@ -381,7 +378,6 @@ private:
   rrc_cfg_t              cfg;
   uint32_t               nof_si_messages;
   asn1::rrc::sib_type2_s sib2;
-  asn1::rrc::sib_type7_s sib7;
 
   void run_thread();
   void rem_user_thread(uint16_t rnti);
