--- srsLTE/lib/src/radio/radio.cc	2021-03-25 21:20:59.980589404 +1100
+++ srsLTE_modified/lib/src/radio/radio.cc	2021-03-22 16:20:01.272532000 +1100
@@ -364,10 +364,10 @@ void radio::set_tx_srate(double srate)
         nsamples = 160;
       } else {
         /* Interpolate from known values */
+        nsamples = cur_tx_srate*(uhd_default_tx_adv_samples * (1/cur_tx_srate) + uhd_default_tx_adv_offset_sec);
         log_h->console(
-            "\nWarning TX/RX time offset for sampling rate %.0f KHz not calibrated. Using interpolated value\n\n",
-            cur_tx_srate);
-        nsamples = cur_tx_srate*(uhd_default_tx_adv_samples * (1/cur_tx_srate) + uhd_default_tx_adv_offset_sec);        
+        	"\nWarning TX/RX time offset for sampling rate %.0f KHz not calibrated. Using interpolated value = %d\n\n",
+        	cur_tx_srate, nsamples);
       }
       
     }else if(!strcmp(srslte_rf_name(&rf_device), "uhd_usrp2")) {
