--- srsLTE/lib/src/common/CMakeLists.txt	2021-03-25 21:20:59.964589254 +1100
+++ srsLTE_modified/lib/src/common/CMakeLists.txt	2021-03-22 16:20:01.268532000 +1100
@@ -22,7 +22,7 @@ file(GLOB CXX_SOURCES "*.cc")
 file(GLOB C_SOURCES "*.c")
 
 add_library(srslte_common STATIC ${C_SOURCES} ${CXX_SOURCES})
-add_custom_target(gen_build_info COMMAND cmake -P ${CMAKE_BINARY_DIR}/SRSLTEbuildinfo.cmake)
+add_custom_target(gen_build_info COMMAND cmake -P ${PROJECT_BINARY_DIR}/SRSLTEbuildinfo.cmake)
 add_dependencies(srslte_common gen_build_info)
 
 add_executable(arch_select arch_select.cc)
