--- srsLTE/lib/src/upper/CMakeLists.txt	2021-03-25 21:20:59.984589442 +1100
+++ srsLTE_modified/lib/src/upper/CMakeLists.txt	2021-03-22 16:20:01.268532000 +1100
@@ -19,6 +19,6 @@
 #
 
 file(GLOB SOURCES "*.cc")
-add_library(srslte_upper STATIC ${SOURCES})
+add_library(srslte_upper STATIC ${SOURCES} ${SRSLTE_EXT_SOURCE})
 target_link_libraries(srslte_upper srslte_common srslte_asn1)
 install(TARGETS srslte_upper DESTINATION ${LIBRARY_DIR})
\ No newline at end of file
