--- srsLTE/lib/src/phy/phch/prach.c	2021-03-25 21:20:59.972589329 +1100
+++ srsLTE_modified/lib/src/phy/phch/prach.c	2021-03-24 22:50:26.983183000 +1100
@@ -523,6 +523,7 @@ int srslte_prach_gen(srslte_prach_t *p,
       ERROR("Error no space for PRACH: frequency offset=%d, N_rb_ul=%d\n", freq_offset, N_rb_ul);
       return ret;
     }
+    DEBUG("srslte_prach_gen: frequency offset=%d, N_rb_ul=%d, K=%d\n", freq_offset, N_rb_ul, K);
 
     DEBUG("N_zc: %d, N_cp: %d, N_seq: %d, N_ifft_prach=%d begin: %d\n",
           p->N_zc, p->N_cp, p->N_seq, p->N_ifft_prach, begin);
