--- srsLTE/srsenb/src/phy/sf_worker.cc	2021-03-25 21:20:59.988589480 +1100
+++ srsLTE_modified/srsenb/src/phy/sf_worker.cc	2021-03-22 16:20:01.252532000 +1100
@@ -25,6 +25,10 @@
 
 #include "srsenb/hdr/phy/sf_worker.h"
 
+#ifdef ENABLE_DDRL
+#include "ops_global_sync.h"
+#endif
+
 #define Error(fmt, ...)                                                                                                \
   if (SRSLTE_DEBUG_ENABLED)                                                                                            \
   log_h->error(fmt, ##__VA_ARGS__)
@@ -354,8 +358,10 @@ void sf_worker::work_imp()
   if (!running) {
     return;
   }
-
   pthread_mutex_lock(&mutex);
+#ifdef ENABLE_DDRL
+    ops_global_sync::ops_report_phy_rx();
+#endif
   is_worker_running = true;
 
   srslte_mbsfn_cfg_t mbsfn_cfg;
