--- srsLTE/srsenb/src/phy/txrx.cc	2021-03-25 21:20:59.988589480 +1100
+++ srsLTE_modified/srsenb/src/phy/txrx.cc	2021-03-24 22:50:26.983183000 +1100
@@ -84,7 +84,7 @@ void txrx::run_thread()
   uint32_t sf_len = SRSLTE_SF_LEN_PRB(worker_com->cell.nof_prb);
   
   float samp_rate = srslte_sampling_freq_hz(worker_com->cell.nof_prb);
-  log_h->console("Setting Sampling frequency %.2f MHz\n", (float) samp_rate/1000000);
+  log_h->console("Setting Sampling frequency %.2f MHz, N_RB %d\n", (float) samp_rate/1000000, worker_com->cell.nof_prb);
 
   // Configure radio
   radio_h->set_rx_srate(0, samp_rate);
