--- srsLTE/srsenb/src/stack/mac/CMakeLists.txt	2021-03-25 21:20:59.992589519 +1100
+++ srsLTE_modified/srsenb/src/stack/mac/CMakeLists.txt	2021-03-22 16:20:01.252532000 +1100
@@ -19,6 +19,11 @@
 #
 
 file(GLOB SOURCES "*.cc")
-add_library(srsenb_mac STATIC ${SOURCES})
+add_library(srsenb_mac STATIC ${SOURCES} ${SRSENB_EXT_SOURCE})
+if(BUILD_DDRL)
+	target_link_libraries(srsenb_mac ops_async_agent)
+else(BUILD_DDRL)
+	message(STATUS "DDRL build disabled")
+endif(BUILD_DDRL)
 install(TARGETS srsenb_mac DESTINATION ${LIBRARY_DIR})
 
