--- srsLTE/srsenb/src/stack/mac/scheduler_ue.cc	2021-03-25 21:20:59.992589519 +1100
+++ srsLTE_modified/srsenb/src/stack/mac/scheduler_ue.cc	2021-03-22 16:20:01.252532000 +1100
@@ -96,7 +96,11 @@ void sched_ue::set_cfg(uint16_t
     Info("SCHED: Added user rnti=0x%x\n", rnti);
     // Config HARQ processes
     for (int i = 0; i < SCHED_MAX_HARQ_PROC; i++) {
-      dl_harq[i].config(i, cfg.maxharq_tx, log_h);
+#ifdef ENABLE_DDRL
+    	dl_harq[i].config(i, 0, log_h);
+#else
+	dl_harq[i].config(i, cfg.maxharq_tx, log_h);
+#endif
       ul_harq[i].config(i, cfg.maxharq_tx, log_h);
     }
 
