--- srsLTE/srsue/src/metrics_csv.cc	2021-03-25 21:21:00.000589594 +1100
+++ srsLTE_modified/srsue/src/metrics_csv.cc	2021-03-22 16:20:01.228531000 +1100
@@ -71,7 +71,7 @@ void metrics_csv::set_metrics(ue_metrics
   pthread_mutex_lock(&mutex);
   if (file.is_open() && ue != NULL) {
     if(n_reports == 0) {
-      file << "time;rsrp;pl;cfo;dl_mcs;dl_snr;dl_turbo;dl_brate;dl_bler;ul_ta;ul_mcs;ul_buff;ul_brate;ul_bler;rf_o;rf_u;rf_l;is_attached\n";
+      file << "time;rsrp;pl;cfo;dl_mcs;dl_snr;dl_turbo;dl_brate;dl_bler;rx_err;rx_pkt;ul_ta;ul_mcs;ul_buff;ul_brate;ul_bler;rf_o;rf_u;rf_l;is_attached\n";
     }
 
     file << (metrics_report_period * n_reports) << ";";
@@ -99,9 +99,12 @@ void metrics_csv::set_metrics(ue_metrics
       rx_errors += metrics.stack.mac[r].rx_errors;
     }
     if (rx_pkts > 0) {
-      file << float_to_string((float)100 * rx_errors / rx_pkts, 1);
+      file << float_to_string((float)100 * rx_errors / rx_pkts, 5);
+	std::cout << "Block Error Rate:" << (float) 100 * rx_errors / rx_pkts << ":" << rx_errors << ":" << rx_pkts << std::endl;
+      file << rx_errors << ";" << rx_pkts << ";";
     } else {
       file << float_to_string(0, 2);
+      file << 0 << ";" << 0 << ";";
     }
 
     file << float_to_string(metrics.phy.sync[0].ta_us, 2);
@@ -133,6 +136,7 @@ void metrics_csv::set_metrics(ue_metrics
     file << float_to_string(metrics.rf.rf_l, 2);
     file << (metrics.stack.rrc.state == RRC_STATE_CONNECTED ? "1.0" : "0.0");
     file << "\n";
+    file.flush();
 
     n_reports++;
   } else {
