--- srsLTE/srsue/src/stack/mac/proc_ra.cc	2021-03-25 21:21:00.000589594 +1100
+++ srsLTE_modified/srsue/src/stack/mac/proc_ra.cc	2021-03-22 16:20:01.236531000 +1100
@@ -179,7 +179,7 @@ void ra_proc::state_response_reception(u
   if (!rar_received) {
     uint32_t interval = srslte_tti_interval(tti, ra_tti + 3 + rach_cfg.responseWindowSize - 1);
     if (interval > 0 && interval < 100) {
-      Error("RA response not received within the response window\n");
+      Error("RA response not received within the response window, interval = %d\n", interval);
       response_error();
     }
   }
