--- srsLTE/lib/include/srslte/upper/rlc.h	2021-03-25 21:20:59.932588950 +1100
+++ srsLTE_modified/lib/include/srslte/upper/rlc.h	2021-03-22 16:20:01.272532000 +1100
@@ -28,7 +28,9 @@
 #include "srslte/interfaces/ue_interfaces.h"
 #include "srslte/upper/rlc_metrics.h"
 #include "srslte/upper/rlc_common.h"
-
+#ifdef ENABLE_DDRL
+#include "srs_ext.h"
+#endif
 namespace srslte {
 
 /****************************************************************************
@@ -41,8 +43,15 @@ class rlc
     :public srsue::rlc_interface_mac
     ,public srsue::rlc_interface_pdcp
     ,public srsue::rlc_interface_rrc
+#ifdef ENABLE_DDRL
+		, public ue_rlc_ext
+#endif
 {
 public:
+#ifdef ENABLE_DDRL
+		long get_head_of_line(uint8_t lc_id);
+		struct timeval get_head_timestamp(uint8_t lc_id);
+#endif
   rlc(log* rlc_log_);
   virtual ~rlc();
   void init(srsue::pdcp_interface_rlc* pdcp_,
